### PR TITLE
optimize l2 cities data

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/TerritoriesFileParser.java [hunk 1]
```diff
@@ -24,14 +24,23 @@ public TerritoryDTO getTerritoriesByLanguage(String language) {
...
+				.get(ConstantsKeys.TERRITORIES);
 		Object defaultRegionCode = JSONUtils.getMapFromJson(regionJson).get(ConstantsKeys.DEFAULT_REGION_CODE);
 		dto.setTerritories(terrMap);
 		dto.setDefaultRegionCode(defaultRegionCode.toString());
+		dto.setCities(null);
+		return dto;
+	}
 
+	@SuppressWarnings("unchecked")
+	public TerritoryDTO getCitysByLanguage(String language) {
```
Comment (huihuiw01) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/TerritoriesFileParser.java:37]: getCitysByLanguage -> getCitiesByLanguage

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java [hunk 2]
```diff
@@ -224,44 +226,59 @@ private String tittleCase(String displayName) {
...
+					TranslationCache3.addCachedObject(CacheName.REGION, TERRITORY_REGIONS + lang, TerritoryDTO.class,
+							cacheTerritoryRegions);
 				}
 			}
-
-			TerritoryDTO territory = cacheTerritory.shallowCopy();
-			if (!StringUtils.isEmpty(territory.getCities()) && Boolean.parseBoolean(displayCity)) {
-				if (!StringUtils.isEmpty(regions)) {
+			if (Boolean.parseBoolean(displayCity)) {
+				TerritoryDTO cacheTerritoryCitys = TranslationCache3.getCachedObject(CacheName.REGION,
```
Comment (huihuiw01) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java:250]: cacheTerritoryCitys -> cacheTerritoryCities

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java [hunk 3]
```diff
@@ -38,6 +38,8 @@
 	private static final String UI_LIST_OR_MENU = "uiListOrMenu";
 	private static final String NO_CHANGES = "no-change";
 	private static final Logger logger = LoggerFactory.getLogger(LocaleService.class.getName());
+	private static final String TERRITORY_REGIONS = "terr-region";
+	private static final String TERRITORY_CITYS = "terr-city";
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java:42]: How about put cities data in a separate cache?

Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java:42]: Use the different cache key prefix

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/TerritoriesFileParser.java (hunk 1 above)
Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/TerritoriesFileParser.java:37]: done

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java (hunk 2 above)
Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java:250]: done

**Review by Xiaochao8 — APPROVED** [review]

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java (hunk 3 above)
Comment (tigershi) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/locale/LocaleService.java:42]: To reduce the cache key size

**Review by Xiaochao8 — APPROVED** [review]

**Review by linr211 — APPROVED** [review]

**Pull request merged**